cmd/migration-managerd/internal/api/workers.go

```diff
@@ -767,6 +767,13 @@ func (d *Daemon) spinUpMigrationEnv(ctx context.Context, inst migration.Instance
 		return fmt.Errorf("Failed to create new instance %q on migration target %q: %w", instanceDef.Name, it.GetName(), err)
 	}
 
+	reverter.Add(func() {
+		err := it.DeleteVM(instanceDef.Name)
```

Comment:
There probably should be some configuration knob to enable/disable this. When a migration fails it's often necessary to exec into the failed environment to look at the worker's log and/or further poke at whatever caused the failure.